packages/integration-boilerplate-node/src/service.ts

```diff
@@ -0,0 +1,342 @@
+        body.variables = variables;
+      }
+
+      if (operationName) {
+        body.operationName = operationName;
+      }
+
+      const url = `https://${
+        this.config.saasquatchAppDomain
+      }/api/v1/${encodeURIComponent(tenantAlias)}/graphql`;
+
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify(body),
+        headers: {
+          Authorization: `Bearer ${apiToken}`,
+          "Content-Type": "application/json",
+        },
+      });
+
+      const json = await response.json();
+
+      if (response.status !== 200) {
+        throw new Error(JSON.stringify(json));
+      }
+
+      if (json.errors) {
+        const errors = json.errors.map((e: any) => e.message).join("\n");
+        throw new Error(errors);
+      }
+
+      return json;
+    } catch (e) {
+      throw new GraphQLError(e.message);
+    }
+  }
+
+  private createExpressServer() {
+    const server = express();
+
+    // Enable compression
+    server.use(compression());
+
+    // Simple request logging middleware
+    server.use((req, _res, next) => {
+      this.logger.info(`${req.method} ${req.url}`);
+      next();
+    });
+
+    // Force HTTPS for all requests
+    if (this.config.enforceHttps) {
+      server.use(enforce.HTTPS({ trustProtoHeader: true }));
+    }
+
+    // Support JSON bodies
+    server.use(
+      express.json({
+        verify: (req, _res, buf) => {
+          // Save the raw body for token validation
+          // NOTE: The default encoding for buf.toString() is UTF-8
+          req.rawBody = buf.toString();
+        },
+      })
+    );
+
+    //  Support application/x-www-form-urlencoded bodies
+    server.use(express.urlencoded({ extended: false }));
+
+    const requireSaaSquatchSignature = createSaasquatchRequestMiddleware(
+      this.auth,
+      this.logger
+    );
+
+    server.post("/webhook", requireSaaSquatchSignature, async (req, res) => {
+      await webhookHandler(
+        req,
+        res,
+        this,
+        this.getTenantScopedGraphQL(req.body.tenantAlias)
+      );
+    });
+
+    server.post("/form", requireSaaSquatchSignature, async (req, res) => {
+      await formHandler(
+        req,
+        res,
+        this,
+        this.getTenantScopedGraphQL(req.body.tenantAlias)
+      );
+    });
+
+    if (this.options?.customRouter) {
+      server.use("/", this.options.customRouter);
+    }
+
+    // Serve the frontend at the root of the server
+    if (this.config.proxyFrontend) {
+      server.use(
+        "/",
+        createProxyMiddleware({
+          target: this.config.proxyFrontend,
+          changeOrigin: true,
+        })
+      );
+    } else {
+      const frontendPath = path.join(
+        require.main!.path,
+        this.config.staticFrontendPath
+      );
+      server.use(express.static(frontendPath));
+      server.get("/*", (_req, res) => {
+        res.sendFile(path.join(frontendPath, this.config.staticFrontendIndex));
```

Comment:
res.sendFile needs error handling, otherwise it can expose information about the file system

![image](https://user-images.githubusercontent.com/39926324/130704826-7c69c916-f1cf-4d14-8649-8967360bde15.png)

Although this only happens if no frontend is built, that might be common and the file path is set by defaults
```
server.get("/*", (_req, res, next) => {
        res.sendFile(path.join(frontendPath, this.config.staticFrontendIndex), undefined, (err)=> {
            if (err) next();
        });
});
```
The above will fix the issue and the error message displayed will be `Cannot GET /`